xrootd/xrdproto/sigver/sigver.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sigver contains the structures describing sigver request.
+package sigver // import "go-hep.org/x/hep/xrootd/xrdproto/sigver"
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+
+	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+	"go-hep.org/x/hep/xrootd/xrdproto/verifyw"
+	"go-hep.org/x/hep/xrootd/xrdproto/write"
+)
+
+// RequestID is the id of the request, it is sent as part of message.
+// See xrootd protocol specification for details: http://xrootd.org/doc/dev45/XRdv310.pdf, 2.3 Client Request Format.
+const RequestID uint16 = 3029
+
+type Flags uint8
+
+const (
+	NoData Flags = 1
+)
+
+// Request holds the sigver request parameters.
+type Request struct {
+	RequestId uint16
+	Version   byte
+	Flags     Flags
+	Seqno     int64
+	Crypto    byte
+	_         [3]byte
+	Signature []byte
+}
+
+// ReqID implements xrdproto.Request.ReqID.
+func (req *Request) ReqID() uint16 { return RequestID }
+
+// ShouldSign implements xrdproto.Request.ShouldSign.
+func (req *Request) ShouldSign() bool { return false }
+
+// MarshalXrd implements xrdproto.Marshaler.
+func (o Request) MarshalXrd(wBuffer *xrdenc.WBuffer) error {
+	wBuffer.WriteU16(o.RequestId)
+	wBuffer.WriteU8(o.Version)
+	wBuffer.WriteU8(uint8(o.Flags))
+	wBuffer.WriteI64(o.Seqno)
+	wBuffer.WriteU8(o.Crypto)
+	wBuffer.Next(3)
+	wBuffer.WriteLen(len(o.Signature))
+	wBuffer.WriteBytes(o.Signature)
+	return nil
+}
+
+// UnmarshalXrd implements xrdproto.Unmarshaler.
+func (o *Request) UnmarshalXrd(rBuffer *xrdenc.RBuffer) error {
+	o.RequestId = rBuffer.ReadU16()
+	o.Version = rBuffer.ReadU8()
+	o.Flags = Flags(rBuffer.ReadU8())
+	o.Seqno = rBuffer.ReadI64()
+	o.Crypto = rBuffer.ReadU8()
+	rBuffer.Skip(3)
+	o.Signature = make([]byte, rBuffer.ReadLen())
+	rBuffer.ReadBytes(o.Signature)
+	return nil
+}
+
+func NewRequest(requestID uint16, seqno int64, data []byte) Request {
+	seqnoData := make([]byte, 8)
```

Comment:
`s/seqnoData := make([]byte, 8)/var seqid [8]byte/`
(to reduce allocation pressure on the GC).

then:
```go
binary.BigEndian.PutUint64(seqid[:], uint64(seqno))
```
and:
```go
hash.Write(seqid[:])
```